ocs_ci/ocs/resources/pod.py

```diff
@@ -69,7 +69,8 @@ def __init__(self, **kwargs):
         """
         self.pod_data = kwargs
         # configure http[s]_proxy env variable, if applicable
-        update_container_with_proxy_env(self.pod_data)
+        if config.DEPLOYMENT.get("disconnected"):
```

Comment:
@vavuthu I think it's not needed as this is handled in the function:
https://github.com/red-hat-storage/ocs-ci/blob/c9a547e069a67708eac2f4c390b32f8d341e0ded/ocs_ci/helpers/proxy.py#L56

Reply:
@dahorak am I right?

Reply:
to genearte http_proxy, it initialize OCP object every time https://github.com/red-hat-storage/ocs-ci/blob/c9a547e069a67708eac2f4c390b32f8d341e0ded/ocs_ci/helpers/proxy.py#L31:L38 when Pod is initialized.

I think we need to get to proxy information from cluster only in disconnected environment.

Below are the logs from vSphere deployment
```
10:59:33 - MainThread - ocs_ci.deployment.deployment - INFO - MDS deployment is successful!
10:59:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get storageclass ocs-storagecluster-ceph-rbd -o yaml
10:59:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod  -n openshift-monitoring -o yaml
10:59:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
10:59:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
10:59:35 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
10:59:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
10:59:36 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
10:59:36 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
10:59:36 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
10:59:36 - MainThread - ocs_ci.ocs.monitoring - INFO - Creating configmap cluster-monitoring-config

11:00:39 - MainThread - ocs_ci.ocs.monitoring - INFO - Verified: Created PVCs are in Bound state.
11:00:39 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod alertmanager-main-0 -n openshift-monitoring -o yaml
11:00:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
11:00:40 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
11:00:40 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
11:00:40 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
11:00:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring rsh alertmanager-main-0 df -kh
11:00:40 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Defaulting container name to alertmanager.
Use 'oc describe pod/alertmanager-main-0 -n openshift-monitoring' to see all of the containers in this pod.

11:00:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod alertmanager-main-1 -n openshift-monitoring -o yaml
11:00:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
11:00:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring rsh alertmanager-main-1 df -kh
11:00:41 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Defaulting container name to alertmanager.
Use 'oc describe pod/alertmanager-main-1 -n openshift-monitoring' to see all of the containers in this pod.

11:00:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod alertmanager-main-2 -n openshift-monitoring -o yaml
11:00:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
11:00:41 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
11:00:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring rsh alertmanager-main-2 df -kh
11:00:42 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Defaulting container name to alertmanager.
Use 'oc describe pod/alertmanager-main-2 -n openshift-monitoring' to see all of the containers in this pod.

11:00:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod prometheus-k8s-0 -n openshift-monitoring -o yaml
11:00:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
11:00:42 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
11:00:42 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
11:00:42 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''
11:00:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring rsh prometheus-k8s-0 df -kh
11:00:42 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Defaulting container name to prometheus.
Use 'oc describe pod/prometheus-k8s-0 -n openshift-monitoring' to see all of the containers in this pod.

11:00:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Pod prometheus-k8s-1 -n openshift-monitoring -o yaml
11:00:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc get Proxy cluster -o yaml
11:00:43 - MainThread - ocs_ci.helpers.proxy - INFO - Using http_proxy: ''
11:00:43 - MainThread - ocs_ci.helpers.proxy - INFO - Using https_proxy: ''
11:00:43 - MainThread - ocs_ci.helpers.proxy - INFO - Using no_proxy: ''

```

Reply:
ah, it's trying to figure out if the proxy exists or not. So yep, then it make more sense to skip it here. @dahorak any objection?

Reply:
Sorry, I didn't spotted this issue and PR earlier. Your solution is not good, because Proxy might be configured not only for Disconnected environment, but also for _Cluster behind proxy_ environment.

I've already created another PR on Friday, where I'm fixing this issue (loading Proxy object only once and lowering log level to debug for some messages): https://github.com/red-hat-storage/ocs-ci/pull/3821.